src/cursor.rs

```diff
@@ -126,6 +126,11 @@ impl<T> RowCursor<T> {
 
             match self.raw.next().await? {
                 Some(chunk) => self.bytes.extend(chunk),
+                None if self.bytes.remaining() > 0 => {
+                    // If some data is left, we have an incomplete row in the buffer.
+                    // This is usually a schema mismatch on the client side.
+                    return Err(Error::NotEnoughData);
+                }
```

Comment:
Note:`Query::fetch_one()` and `Query::fetch_optional()` can still finish successfully even in the case of schema mismatch, because they don't read the whole response. This is trade-off made ages ago to avoid parsing unused bytes.

Should we also revise that behavior? By the way, if the server returns a lot of rows (or large ones), it's possible to get `EPIPE` errors on the server side (because not all bytes are delivered).

We have several options here:
1. Parse all rows (better to detect the schema mismatch) and discard all besides the first one.
2. Don't parse them, but read the whole stream (to avoid `EPIPE` on the CH side).
3. Leave as-is, wait for `Native`/`RowBinaryWithNAT` for reliable error-detection.

Reply:
Let's discuss a possible follow-up in https://github.com/ClickHouse/clickhouse-rs/issues/187